packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -177,7 +177,7 @@ class _WindowsUtils extends OperatingSystemUtils {
       if (entity is! File) {
         continue;
       }
-      final File file = entity;
+      final File file = entity as File;
```

Comment:
hm, this is similar to the case above. i guess i was wrong about `is!`. does it let you avoid the `as` if you wrap instead of bailing? maybe file a bug on the sdk for this, the compiler should be able to determine that `entity` is a `File` here.

Reply:
The canonical issue is https://github.com/dart-lang/sdk/issues/25565. better type promotion is a requirement of NNBD, and last I heard was a work in progress